receiver/valkeyreceiver/receiver.go

```diff
@@ -0,0 +1,102 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package valkeyreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/valkeyreceiver"
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/valkey-io/valkey-go"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/receiver"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/valkeyreceiver/internal/metadata"
+)
+
+type valkeyScraper struct {
+	client     client
+	settings   component.TelemetrySettings
+	cfg        *Config
+	mb         *metadata.MetricsBuilder
+	configInfo configInfo
+}
+
+func newValkeyScraper(cfg *Config, settings receiver.Settings) (*valkeyScraper, error) {
+	configInfo, err := newConfigInfo(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	return &valkeyScraper{
+		cfg:        cfg,
+		settings:   settings.TelemetrySettings,
+		mb:         metadata.NewMetricsBuilder(cfg.MetricsBuilderConfig, settings),
+		configInfo: configInfo,
+	}, nil
+}
+
+func (vs *valkeyScraper) shutdown(context.Context) error {
+	if vs.client != nil {
+		return vs.client.close()
+	}
+	return nil
+}
+
+func (vs *valkeyScraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
+	if vs.client == nil {
+		opts := valkey.ClientOption{
+			InitAddress: []string{vs.cfg.Endpoint},
+		}
+
+		var err error
+		if opts.TLSConfig, err = vs.cfg.TLS.LoadTLSConfig(context.Background()); err != nil {
+			return pmetric.Metrics{}, err
+		}
+		vs.client, err = newValkeyClient(opts)
```

Comment:
If we're using the `if ... err != nil` pattern with the TLS load, we should use it again here.

```suggestion
		if vs.client, err = newValkeyClient(opts); err != nil {
```

Reply:
Done https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/37005/commits/fc0f24bdd1fcd9a5173c2a447e02570ec051ea66